ipaserver/plugins/certmap.py

```diff
@@ -389,3 +393,115 @@ def execute(self, cn, **options):
+
+        Call Users.ListByCertificate interface and return a list of names
+        corresponding to the users matching the provided cert
+        :param cert: DER cert
+        :raise RemoteRetrieveError: if DBus error occurs
+        """
+        try:
+            pem = x509.make_pem(base64.b64encode(cert))
+            # bug 3306 in sssd returns 0 entry when max_entries = 0
+            # Temp workaround is to use a non-null value, not too high
+            # to avoid reserving unneeded memory
+            max_entries = dbus.UInt32(100)
+            user_paths = self._users_iface.ListByCertificate(pem, max_entries)
+            users = list()
+            for user_path in user_paths:
+                user_obj = self._bus.get_object(DBUS_SSSD_NAME, user_path)
+                user_iface = dbus.Interface(user_obj, DBUS_PROPERTY_IF)
+                user_login = user_iface.Get(DBUS_SSSD_USER_IF, 'name')
+                users.append(user_login)
+            return users
+        except dbus.DBusException as e:
+            err_name = e.get_dbus_name()
+            # If there is no matching user, do not consider this as an
+            # exception and return an empty list
+            if err_name == 'org.freedesktop.sssd.Error.NotFound':
+                return list()
+            self.log.error(
+                'Failed to use interface {iface}. DBus '
+                'exception is {exc}.'.format(iface=DBUS_SSSD_USERS_IF, exc=e))
+            raise errors.RemoteRetrieveError(
+                reason=_('Failed to find users over SystemBus. '
+                         ' See details in the error_log'))
+
+
+@register()
+class certmap_match(Command):
+    __doc__ = _('Search for users matching the provided certificate.')
+
+    has_output = (
+        output.summary,
+        output.Output('uids', (list, tuple, type(None)), _('Matched uid')),
+        output.Output('count', int, _('Number of entries returned')),
+        )
+
+    takes_options = (
+        Bytes(
+            'certificate', validate_certificate,
+            cli_name='certificate',
+            label=_('Certificate'),
+            doc=_('Base-64 encoded user certificate')
+        ),
+    )
+
+    def execute(self, *args, **options):
+        """
+        Search for users matching the provided certificate.
+
+        The search is performed using SSSD's DBus interface
+        Users.ListByCertificate.
+        SSSD does the lookup based on certificate mapping rules, using
+        FreeIPA domain and trusted domains.
+        :raise RemoteRetrieveError: if DBus returns an exception
+        """
+        sssd = _sssd(self.log)
+        result = dict()
+
+        cert = options['certificate']
+        users = sssd.list_users_by_cert(cert)
+        result['count'] = len(users)
+        result['uids'] = users
+        result['summary'] = u'{count} user{plural} matched'.format(
+            count=len(users), plural='' if len(users) == 1 else 's')
```

Comment:
This is not translatable. Instead of filling in `summary` yourself, you should specify the `msg_summary` class attribute and have it filled automatically.